### PR TITLE
--outfile options

### PR DESCRIPTION
**Pull request by PythaGorilla**
added --outfile option

### CODE REVIEW

**Comment by chezou** [comment]
Thanks for your contributing! I'm sorry but I applyed `flake8` on master. https://github.com/chezou/tabula-py/commit/b3ff1596c409091225aa4aad4b631282a74a744c
Could you rebase on it?

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,12 @@ target/
 
 .ipynb_checkpoints/
 .python-version
+
+#Pycharm settings
```
Comment (chezou) [inline .gitignore]: I don't think these Pycharm specific setting should be included.

File: tabula/wrapper.py [hunk 2]
```diff
@@ -8,74 +8,84 @@
...
+    Args:
+      input_path (str): File path of tareget PDF file.
+      options (str, optional): Option string for tabula-java.
+      pages (str, int, :obj:`list` of :obj:`int`, optional): An optional values specifying pages to extract from. It allows `str`, `int`, :obj:`list` of :obj:`int`. Example: '1-2,3', 'all' or [1,2]
+      guess (bool, optional): Guess the portion of the page to analyze per page.
+      area (:obj:`list` of :obj:`float`, optional): Portion of the page to analyze(top,left,bottom,right). Example: [269.875,12.75,790.5,561]. Default is entire page
+      spreadsheet (bool, optional): Force PDF to be extracted using spreadsheet-style extraction (if there are ruling lines separating each cell, as in a PDF of an Excel spreadsheet)
+      nospreadsheet (bool, optional): Force PDF not to be extracted using spreadsheet-style extraction (if there are ruling lines separating each cell, as in a PDF of an Excel spreadsheet)
+      password (bool, optional): Password to decrypt document. Default is empty
+      silent (bool, optional): Suppress all stderr output.
```
Comment (chezou) [inline tabula/wrapper.py]: Could you add description about `outfile` which you added?

File: tabula/wrapper.py [hunk 3]
```diff
@@ -8,74 +8,84 @@
...
 
-  args = ["java", "-jar", jar_path] + __options + [input_path]
+    if outfile:
+        __options += ["--outfile", outfile]
 
-  output = subprocess.check_output(
-    args,
-    stderr=subprocess.STDOUT)
+    args = ["java", "-jar", jar_path] + __options + [input_path]
+    stdoutput = subprocess.check_output(
```
Comment (chezou) [inline tabula/wrapper.py]: What is your intention to change this variable name?

File: tabula/wrapper.py [hunk 4]
```diff
@@ -8,74 +8,84 @@
...
 
-  # parse options
-  if pages:
-    __pages = pages
-    if type(pages) == int:
-      __pages = str(pages)
-    elif type(pages) in [list, tuple]:
-      __pages = ",".join(map(str, pages))
+    # parse options
+    if pages:
```
Comment (chezou) [inline tabula/wrapper.py]: Many problems related to indentation will be solved after rebasing master.

File: tabula/wrapper.py (hunk 3 above)
Comment (PythaGorilla) [inline tabula/wrapper.py]: I think output and outfile are ambiguous when placed together. So i changed output to stdoutput to clarify

File: .gitignore (hunk 1 above)
Comment (PythaGorilla) [inline .gitignore]: Already removed these lines in latest commits, should be clean by now. I messed up the commit log a little bit. Just check the latest.

**Comment by chezou** [comment]
If you want to avoid parsing error with this change, you can try updated version of tabula-py. It was fixed critical issue.

**Comment by chezou** [comment]
I created `convert_into()` method. https://github.com/chezou/tabula-py/commit/5b6eae85ce80b359fd7cec6150f01f450f291156 I wonder if it is useful for your usecase.

**Pull request closed without merging**